src/sequence_manager.hpp

```diff
@@ -71,8 +94,34 @@ class SequenceManager {
 
     Status removeSequence(const uint64_t sequenceId);
 
-    Status removeTimedOutSequences(std::chrono::steady_clock::time_point currentTime);
+    Status checkForTimedOutSequences();
```

Comment:
Why changing the method name? It seems now that we are actually removing timed out sequences in that method so previous name looks like a better fit to me.